selfdrive/thermald/thermald.py

```diff
@@ -403,8 +377,10 @@ def thermald_thread():
         started_seen = True
         os.system('echo performance > /sys/class/devfreq/soc:qcom,cpubw/governor')
     else:
+      if startup_conditions["ignition"] and (startup_conditions != startup_conditions_prev):
+        cloudlog.event("Startup blocked", startup_conditions=startup_conditions)
       if should_start_prev or (count == 0):
-        put_nonblocking("IsOffroad", "1")
+        params.put("IsOffroad", "1")
```

Comment:
Are you sure this is okay to do nonblocking? I agree it's offroad, but params.put has been the source of so many issues. (perhaps nonblocking should be the default in later versions)

Reply:
It shouldn't be a problem since it's only done once on the onroad -> offroad transition. It's done to prevent an unlikely race condition with things like snapshot, which could start a second camerad.